src/Handler.php

```diff
@@ -88,22 +88,30 @@ public function onPostCmdEvent(\Composer\Script\Event $event) {
     // Only install the scaffolding if drupal/core was installed,
     // AND there are no scaffolding files present.
     if (isset($this->drupalCorePackage)) {
-      $this->downloadScaffold();
+      $this->downloadScaffold($event->isDevMode());
       // Generate the autoload.php file after generating the scaffold files.
       $this->generateAutoload();
     }
   }
 
   /**
    * Downloads drupal scaffold files for the current process.
+   *
+   * @param bool $dev
+   *   TRUE if dev packages are installed. FALSE otherwise.
    */
-  public function downloadScaffold() {
+  public function downloadScaffold($dev = FALSE) {
```

Comment:
This is also called from (Plugin.php)[https://github.com/drupal-composer/drupal-scaffold/blob/2.3.0/src/Plugin.php#L77).

Rather than risking changing the API of this method every time we have an enhancement to the state we respond to, it would probably be better to pass the `$event` in here, and let `downloadScaffold()` access `$event->isDevMode()` itself.